doc/source/examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability.rst

```diff
@@ -0,0 +1,252 @@
+
+.. code-block:: text
+    
+    #---------------------------------------------------
+    # Source term
+    #---------------------------------------------------
+    subsection source term
+        set enable                      = true
+        subsection xyz
+            set Function expression     = 0 ; -9.81 ; 0
+        end
+    end
+
+
+The ``physical properties`` subsection defines the physical properties of the fluid. In this example, we need two fluids with densities of 100 and 300 and with an equal kinematic viscosity (0.00153).
+
+
+.. code-block:: text
+
+    #---------------------------------------------------
+    # Physical Properties
+    #---------------------------------------------------
+    subsection physical properties
+        set number of fluids     = 2
+        subsection fluid 0
+          set density              	= 100
+          set kinematic viscosity  	= 0.00153
+        end
+        subsection fluid 1
+          set density 		= 300
+          set kinematic viscosity 	= 0.00153
+        end
+    end
+
+
+In the ``initial condition`` subsection, we need to define the interface between the heavy and light fluids. We define this interface by using a function expression in the ``VOF`` subsection of the ``initial condition``.
+
+
+.. code-block:: text
+
+    #---------------------------------------------------
+    # Initial Condition
+    #---------------------------------------------------
+    subsection initial conditions
+        set type = nodal
+        subsection uvwp
+                set Function expression = 0; 0; 0
+        end
+        
+        subsection VOF
+                set Function expression = if (y>(0.5 + 0.1 * 0.25 * cos(2 *3.1415 * x / 0.25)) , 1, 0)
+        end
+    end
+
+
+The ``initial refinement`` of the mesh is equal to 7, but we use a mesh adaptation to coarsen the mesh in cells far from the interface to improve the computation performance. Here, we choose ``phase`` as the ``refinement variable`` and 5 as the ``min refinement level``. 
+
+
+.. code-block:: text
+
+    #---------------------------------------------------
+    # Mesh Adaptation
+    #---------------------------------------------------
+    subsection mesh adaptation
+      set type                    	= kelly
+      set variable                	= phase
+      set fraction type           	= fraction
+      set max refinement level    	= 7
+      set min refinement level    	= 5
+      set frequency               	= 1
+      set fraction refinement     	= 0.9
+      set fraction coarsening     	= 0.001
+    end
+
+
+In the ``VOF`` subsection, we enable ``interface sharpening`` to reconstruct the interface and keep it sharp during the simulation. Note that here we use the ``constant`` method for interface sharpening. We can also use ``adaptative`` type for interface sharpening, but the ``mass conservation`` results show that choosing a ``constant`` method does not affect the mass conservation significantly.
```

Comment:
Yeah I would really add another small section to this example where you do the same thing with adaptive mass conservation. It would be the ideal place to show it.

Reply:
Added.